### ISSUE TITLE
references package even if no cited functions/methods used

### ISSUE THREAD

**Issue by yarikoptic** [duecredit/duecredit#48]
@mvdoc Why sklearn listed (not that it shouldn't since we use its functionality here but none of the decorated methods was called AFAIK)?
```
$> python -m duecredit examples/example_scipy.py
I: Simulating 4 blobs           
I: Done clustering 4 blobs

DueCredit Report:
- scipy (v 0.14.1) [1]
  - scipy.cluster.hierarchy:linkage (v 0.14.1) [2]
- sklearn (v 0.16.1) [3]

2 packages cited
0 modules cited
1 functions cited

References
----------

[1] Jones, E. et al., 2001. SciPy: Open source scientific tools for Python.
[2] Sibson, R., 1973. SLINK: an optimally efficient algorithm for the single-link cluster method. The Computer Journal, 16(1), pp.30–34.
[3] Pedregosa, F. et al., 2011. Scikit-learn: Machine learning in Python. The Journal of Machine Learning Research, 12, pp.2825–2830.
```

**Comment by mvdoc** [comment]
well -- why not? we use it, so it's referenced :-)

**Comment by yarikoptic** [comment]
but we are not using any  duecredited function...  how do we know that we don't just import it?

**Comment by mvdoc** [comment]
that's fair -- I will look into it

**Comment by mvdoc** [comment]
So I'm looking into this, AFAIU the issue comes from injector.py that is set up to always import the package if the module/functions are imported...but I don't know what would be the safest way to change it since I have to digest the code a bit more. Any suggestion? Here's the relevant debugging output

```
2015-09-03 16:44:09,730 [DEBUG] Processing delayed injection for sklearn (injector.py:141)
2015-09-03 16:44:09,730 [Level 3] Importing duecredit.injections.mod_sklearn (injector.py:146)
2015-09-03 16:44:09,730 [DEBUG] Performing undecorated import of entries (injector.py:237)
2015-09-03 16:44:09,730 [Level 3] Calling injector of duecredit.injections.mod_sklearn (injector.py:157)
2015-09-03 16:44:09,730 [DEBUG] Adding citation entry BibTeX('@article{pedregosa2... for sklearn:None (injector.py:123)
2015-09-03 16:44:09,730 [DEBUG] Adding citation entry Doi('10.1126/science.113680... for sklearn.cluster.affinity_propagation_:None (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Adding citation entry Doi('10.1.1.140.301', key='... for sklearn.cluster.bicluster:SpectralCoclustering (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Adding citation entry Doi('10.1.1.135.1608', key=... for sklearn.cluster.bicluster:SpectralBiclustering (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Adding citation entry Doi('10.1145/233269.233324'... for sklearn.cluster.birch:Birch (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Adding citation entry Url('https://code.google.co... for sklearn.cluster.birch:Birch (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Adding citation entry BibTeX('@inproceedings{este... for sklearn.cluster.dbscan_:dbscan (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Adding citation entry Doi('10.1109/34.1000236', k... for sklearn.cluster.mean_shift_:MeanShift (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Adding citation entry Doi('10.1109/ICCV.2003.1238... for sklearn.cluster.spectral:discretize (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Adding citation entry Doi('10.1109/34.868688', ke... for sklearn.cluster.spectral:spectral_clustering (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Adding citation entry Doi('10.1007/s11222-007-903... for sklearn.cluster.spectral:spectral_clustering (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Adding citation entry Doi('10.1023/A:101093340432... for sklearn.ensemble.forest:RandomForestClassifier.predict_proba (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Adding citation entry Doi('10.1023/A:101093340432... for sklearn.ensemble.forest:RandomForestRegressor.predict (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Adding citation entry BibTeX('@BOOK{breiman-fried... for sklearn.tree.tree:DecisionTreeClassifier.predict_proba (injector.py:123)
2015-09-03 16:44:09,731 [DEBUG] Process 1 injections for module sklearn (injector.py:175)
2015-09-03 16:44:09,731 [Level 4] Considering 1 records for decoration of None:None (injector.py:197)
2015-09-03 16:44:09,731 [Level 1] Collector added entry pedregosa2011scikit (collector.py:204)
2015-09-03 16:44:09,732 [Level 3] Citing directly None:None since obj_path is empty (injector.py:207)
2015-09-03 16:44:09,732 [Level 3] Done processing injections for module sklearn (injector.py:210)
2015-09-03 16:44:09,732 [Level 1] Returning <module 'sklearn.datasets' from '/usr/local/lib/python2.7/site-packages/sklearn/datasets/__init__.pyc'> (injector.py:276)
```

**Comment by yarikoptic** [comment]
the analysis logic should happen during output, as it was before somewhat -- if there are no references triggered for submodules -- do not output that module reference (unless cite_module=True for that module, as we do for numpy only atm)

**Comment by mvdoc** [comment]
Gotcha. Thanks for the hint

**Comment by yarikoptic** [comment]
closed by #86
